### PR TITLE
Mercurial support, more complete

### PR DESCRIPTION
**Pull request by odcinek**

### CODE REVIEW

File: lib/between_meals/repo.rb [hunk 1]
```diff
@@ -50,89 +50,89 @@ def self.get(type, repo_path, logger)
     end
 
     def exists?
-      fail 'Not implemented'
+      fail "#{__method__} not implemented"
```
Comment (jaymzh) [inline lib/between_meals/repo.rb]: The fail message will include the function that threw the function, so this is redundant, no?

Comment (odcinek) [inline lib/between_meals/repo.rb]: No it won't

File: lib/between_meals/repo/git.rb [hunk 2]
```diff
@@ -58,7 +58,7 @@ def last_author
       end
 
       def head_parents
-        @repo.head.target.parents
+        @repo.head.target.parents.map {|x| {:time => x.time}}
```
Comment (jaymzh) [inline lib/between_meals/repo/git.rb]: er, this drops the actual revision, no?

File: lib/between_meals/repo/git.rb [hunk 3]
```diff
@@ -138,7 +138,7 @@ def _config key
         cmd = Mixlib::ShellOut.new("#{@bin} config #{key}")
         cmd.run_command
         cmd.stdout.strip
-      rescue
+      rescue => e
```
Comment (jaymzh) [inline lib/between_meals/repo/git.rb]: why? you're not using it.

File: lib/between_meals/repo/git.rb (hunk 2 above)
Comment (odcinek) [inline lib/between_meals/repo/git.rb]: It's not needed, not used anywhere as far as I can tell. This file needs refactoring btw, another sweep

File: lib/between_meals/repo/git.rb (hunk 3 above)
Comment (odcinek) [inline lib/between_meals/repo/git.rb]: true

Comment (odcinek) [inline lib/between_meals/repo/git.rb]: I'm gonna clean this file up in separate commit, along with refactoring and rubocop

File: lib/between_meals/repo.rb (hunk 1 above)
Comment (jaymzh) [inline lib/between_meals/repo.rb]: how so?
```
irb(main):001:0> def fucker
irb(main):002:1>   fail 'ding dong'
irb(main):003:1> end
=> nil
irb(main):004:0> fucker
RuntimeError: ding dong
	from (irb):2:in `fucker'
	from (irb):4
```

File: lib/between_meals/repo/git.rb (hunk 2 above)
Comment (jaymzh) [inline lib/between_meals/repo/git.rb]: we're not the only people writing plugins. It's one thing to change the API, another to drop something the API provides.

File: lib/between_meals/repo/git.rb (hunk 3 above)
Comment (jaymzh) [inline lib/between_meals/repo/git.rb]: k, but don't add crap to cleanup later for no reason :)

File: lib/between_meals/repo/git.rb (hunk 2 above)
Comment (odcinek) [inline lib/between_meals/repo/git.rb]: the api is broken as it is - it's returning a rugged objects

Comment (jaymzh) [inline lib/between_meals/repo/git.rb]: I'm not arguing that. I'm saying we have git users who are writing plugins and who are probably assuming stupid crap. You need to at least provide the hash/rev ID.

File: lib/between_meals/repo.rb (hunk 1 above)
Comment (odcinek) [inline lib/between_meals/repo.rb]: Sigh. We're catching it somewhere higher and hiding the stacktrace, which needs to change, but not now

Comment (jaymzh) [inline lib/between_meals/repo.rb]: hmm. fair enough.

File: lib/between_meals/repo/hg.rb [hunk 4]
```diff
@@ -97,6 +101,44 @@ def files
         end
       end
 
+      def _show
```
Comment (jaymzh) [inline lib/between_meals/repo/hg.rb]: shouldn't this just be private instead of `_`?

**Pull request merged**